src/test/java/de/bonndan/nivio/appuser/AppUserRepositoryTest.java

```diff
@@ -0,0 +1,46 @@
+package de.bonndan.nivio.appuser;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@DataJpaTest
+class AppUserRepositoryTest {
+
+    @Autowired
+    private AppUserRepository appUserRepository;
+
+    AppUserService appUserService = new AppUserService(appUserRepository);
+
+
+    @Test
+    void findByExternalId() {
+        Optional<AppUser> appUser = Optional.of(new AppUser());
```

Comment:
You would use a simple `AppUser` instance here, no need to wrap it in a `Optional`.

Reply:
But for line 33:
`        final Optional<AppUser> fetchedAppUser = appUserRepository.findByExternalId(appUser.get().getExternalId());`

I need appUser to be an Optional.